wagtail/admin/templates/wagtailadmin/pages/listing/_button_with_dropdown.html

```diff
@@ -10,7 +10,7 @@
         <ul class="c-dropdown__menu u-toggle  u-arrow u-arrow--tl u-background">
         {% for button in buttons %}
             <li class="c-dropdown__item ">
-                <a href="{{ button.url }}" {% if button.attrs.filter %}data-filter={{button.attrs.filter}}{% endif %} aria-label="{{ button.attrs.title }}" class="u-link is-live {{ button.classes|join:' ' }}">
+                <a href="{{ button.url }}" {% if button.attrs.filter %}data-bulk-action-filter={{button.attrs.filter}}{% endif %} aria-label="{{ button.attrs.title }}" class="u-link is-live {{ button.classes|join:' ' }}">
```

Comment:
This is the only place I am not 100% sure, now that I see the changes in a small PR.
button with dropdown could be used outside of the bulk actions context.

suggestion: make it accept an attrs.data dict with key for the data attribute and value

```python
attrs = {
    "data": {"bulk-action-filter": "something"}
}
```
and here `{% if button.attrs.data %}{% for data_key, data_value in button.attrs.data.items %} ...`